rust/worker/src/blockstore/types.rs

```diff
@@ -277,53 +280,53 @@ impl<
 
     // TODO: make prefix &str
     pub(crate) fn get_by_prefix(
-        &self,
-        prefix: String,
-    ) -> Result<Vec<(&str, &K, &V)>, Box<dyn ChromaError>> {
+        &'referred_data self,
+        prefix: &'referred_data str,
```

Comment:
we don't need the prefix lifetime here to be 'referred_data do we? Can just be &str like get() i think?

Reply:
If we want to return a `&str` we have to return this very same prefix `&str` (since we can't construct a `String` and return a reference to it from within the `reader_writer.rs` functions I think?). Because of this, the functions in `reader_writer.rs` ask for lifetime annotations on the `&str`s that they accept, which means we need them here too.

Reply:
I'm very open to being wrong -- lifetimes are still not fully intuitive to me (yet!)

Reply:
For posterity: I was wrong :^)